### PR TITLE
mp-input-group component

### PR DESCRIPTION
**Pull request by jbwyme**
http://cometapp.us/c/oQjt3DW2

### CODE REVIEW

File: lib/components/input-group/index.jade [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      - inputAttrs.disabled = true
+    input.mp-input-group-input(
+      oninput=$helpers.setValue
+      onkeyup=$helpers.submitIfEnter
+      attributes=inputAttrs
+    )
+    mp-button.mp-button-primary.mp-input-group-btn(
+      onclick=$helpers.submit
+      attributes={
+        disabled: disabled || value === '',
```
Comment (tdumitrescu) [inline lib/components/input-group/index.jade]: Should we trim `value` to keep the button disabled for all-whitespace input?

File: lib/components/input-group/index.jade [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        'style-override': '.mp-input-group-btn.mp-button-container { border-top-left-radius: 0px; border-bottom-left-radius: 0px; width: 90px; }',
+      }
+    )
+      if $component.isAttributeEnabled('saving')
+        .mp-input-group-spinner
+          include ./spinner.svg
```
Comment (tdumitrescu) [inline lib/components/input-group/index.jade]: pretty generic spinner - can we get it into the standard `<svg-icon>` set?

File: lib/components/input-group/index.jade [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      else
+        | #{$component.getAttribute('label') || 'Save'}
```
Comment (tdumitrescu) [inline lib/components/input-group/index.jade]: interpolating an entire line always seems weird to me. is this different than:
```jade
= $component.getAttribute('label') || 'Save'
```

File: lib/components/input-group/index.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    return this.state.value;
+  }
+
+  set value(value) {
+    this.update({value});
+  }
+
+  attachedCallback() {
+    super.attachedCallback(...arguments);
+    const updates = {};
```
Comment (tdumitrescu) [inline lib/components/input-group/index.js]: `updates` isn't actually used is it? (shouldn't a linter be picking this up?)

File: lib/components/input-group/index.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    if (this.getAttribute('value')) {
+      updates.value = this.getAttribute('value');
+      this.update({value: this.getAttribute('value')});
+    }
+  }
+
+  detachedCallback() {
+    super.detachedCallback(...arguments);
+  }
```
Comment (tdumitrescu) [inline lib/components/input-group/index.js]: i guess this can be removed?

File: lib/components/input-group/index.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+  attributeChangedCallback(name) {
+    super.attributeChangedCallback(...arguments);
+    if (this.initialized && name === 'value') {
+      this.update({value: this.getAttribute('value')});
+    }
```
Comment (tdumitrescu) [inline lib/components/input-group/index.js]: can extract this and the version in `attachedCallback` into a single method like `updateValueFromAttr()`

File: lib/components/input-group/index.jade (hunk 2 above)
Comment (jbwyme) [inline lib/components/input-group/index.jade]: up to @jayjo

**Pull request closed without merging**